atomic_reactor/plugin.py

```diff
@@ -59,6 +56,22 @@ def __init__(self, *args, **kwargs):
         self.args = args
         self.kwargs = kwargs
 
+    @property
+    @abstractmethod
+    def key(self) -> str:
+        """Unique plugin identification
+
+        Output of this plugin can be found in results specified with this key,
+        same thing goes for input: use this key for providing input for this plugin
+
+        Implementation notes: because this must be defined in each plugin it's abstract
+        property. For easy implementation it's just instance property not a class
+        property (with py<=3.8 it requires metaprogramming).
```

Comment:
How is it done with py>3.8?

Reply:
https://stackoverflow.com/questions/128573/using-property-on-classmethods/64738850#64738850

Reply:
I added it into comment.
I moved implementation details from docstring to just comment, IMO it shouldn't be in docstring and exported to documnetation

Reply:
+1 (I don't see the changes yet, though :smile:)

Reply:
because I'm stupid that's why

pushing